packages/loopring_v3/contracts/aux/bridge/Bridge.sol

```diff
@@ -0,0 +1,618 @@
+    function withdrawFromPendingBatchDeposit(
+        uint                              batchID,
+        InternalBridgeTransfer[] calldata transfers,
+        uint[]                   calldata indices
+        )
+        external
+    {
+        // Check if withdrawing from these transfers is possible
+        bytes32 hash = _hashTransfers(batchID, transfers);
+        require(_arePendingTransfersTooOld(hash), "TRANSFERS_NOT_TOO_OLD");
+
+        for (uint i = 0; i < indices.length; i++) {
+            uint idx = indices[i];
+
+            require(!withdrawn[hash][idx], "ALREADY_WITHDRAWN");
+            withdrawn[hash][idx] = true;
+
+            address tokenAddress = exchange.getTokenAddress(transfers[idx].tokenID);
+
+            _transferOut(
+                tokenAddress,
+                transfers[idx].amount,
+                transfers[idx].owner
+            );
+        }
+    }
+
+    // Can be used to withdraw funds that were already deposited to the bridge,
+    // but need to be returned to be able to withdraw from old pending transfers.
+    function forceWithdraw(address[] calldata tokens)
+        external
+        payable
+    {
+        for (uint i = 0; i < tokens.length; i++) {
+            exchange.forceWithdraw{value: msg.value / tokens.length}(
+                address(this),
+                tokens[i],
+                accountID
+            );
+        }
+    }
+
+    function setConnectorTrusted(
+        address connector,
+        bool    trusted
+        )
+        external
+        onlyOwner
+    {
+        trustedConnectors[connector] = trusted;
+        emit ConnectorTrusted(connector, trusted);
+    }
+
+    receive()
+        external
+        payable
+    {}
+
+    // --- Internal functions ---
+
+    function _processTransfers(
+        Context         memory ctx,
+        TransferBatch[] memory operations,
+        bytes           memory txsData
+        )
+        internal
+    {
+        for (uint o = 0; o < operations.length; o++) {
+            InternalBridgeTransfer[] memory transfers = operations[o].transfers;
+
+            // Check if these transfers can be processed
+            bytes32 hash = _hashTransfers(operations[o].batchID, transfers);
+            require(!_arePendingTransfersTooOld(hash), "TRANSFERS_TOO_OLD");
+
+            // Mark transfers as completed
+            pendingTransfers[hash] = 0;
+
+            // Verify transfers
+            TransferTransaction.Transfer memory transfer;
+            for (uint i = 0; i < transfers.length; i++) {
+                TransferTransaction.readTx(txsData, ctx.txIdx++ * ExchangeData.TX_DATA_AVAILABILITY_SIZE, transfer);
+
+                uint16 tokenID = transfers[i].tokenID;
+                require(
+                    transfer.fromAccountID == ctx.accountID &&
+                    transfer.to == transfers[i].owner &&
+                    transfer.from == address(this) &&
```

Comment:
Yes, could be removed, checking the accountID is enough. The address will always be part of the transfer DA though because it's a conditional transfer.